src/hooks/routing/useRouterTrade.ts

```diff
@@ -40,94 +39,60 @@ export function useRouterTrade<TTradeType extends TradeType>(
   gasUseEstimateUSD?: CurrencyAmount<Token>
 } {
   const { provider } = useWeb3React()
-  const isWindowVisible = useIsWindowVisible()
-  // Debounce is used to prevent excessive requests to SOR, as it is data intensive.
-  // Fast user actions (ie updating the input) should be debounced, but currency changes should not.
-  const [debouncedAmount, debouncedOtherCurrency] = useDebounce(
-    useMemo(() => [amountSpecified, otherCurrency], [amountSpecified, otherCurrency]),
-    200
-  )
-  const isDebouncing = amountSpecified !== debouncedAmount && otherCurrency === debouncedOtherCurrency
-  const debouncedAmountSpecified = isWindowVisible ? debouncedAmount : undefined
-
-  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
-    () =>
-      isExactInput(tradeType)
-        ? [debouncedAmountSpecified?.currency, debouncedOtherCurrency]
-        : [debouncedOtherCurrency, debouncedAmountSpecified?.currency],
-    [debouncedAmountSpecified, debouncedOtherCurrency, tradeType]
-  )
-
+  const [currencyIn, currencyOut] = isExactInput(tradeType)
+    ? [amountSpecified?.currency, otherCurrency]
+    : [otherCurrency, amountSpecified?.currency]
   const queryArgs = useRouterArguments({
     tokenIn: currencyIn,
     tokenOut: currencyOut,
-    amount: debouncedAmountSpecified,
+    amount: amountSpecified,
     tradeType,
     routerUrl,
     provider: provider as JsonRpcProvider,
   })
 
-  const { isError, data, currentData } = useGetQuoteQuery(queryArgs ?? skipToken, {
-    // Price-fetching is informational and costly, so it's done less frequently.
+  // Get the cached state *immediately* to update the UI without sending a request - using useGetQuoteQueryState -
+  // but debounce the actual request - using useLazyGetQuoteQuery - to avoid flooding the router / JSON-RPC endpoints.
+  const { isError, data, currentData } = useGetQuoteQueryState(queryArgs)
+  const isValidBlock = useIsValidBlock(Number(data?.blockNumber))
+  const isSyncing = currentData !== data || !isValidBlock
+
+  const [trigger] = useLazyGetQuoteQuery({
+    // PRICE fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`,
   })
-
-  const quoteResult: GetQuoteResult | undefined = useIsValidBlock(Number(data?.blockNumber) || 0) ? data : undefined
+  useEffect(() => {
```

Comment:
maybe would help readability/reusability to move this behavior to a `useTimeout` hook similar to https://github.com/Uniswap/interface/blob/f9fc506db4552af87f6f619fa0c1ec7b3dd08619/src/lib/hooks/useInterval.ts